dotnet/src/Experimental/Process.LocalRuntime/LocalProcess.cs

```diff
@@ -324,6 +324,24 @@ private void EnqueueExternalMessages(Queue<ProcessMessage> messageChannel)
         }
     }
 
+    /// <summary>
+    /// Process public events to emit them externally at the same time as the internal events
+    /// </summary>
+    /// <param name="events">events to be used to process only the public events</param>
+    /// <returns>boolean with details if there were public events detected</returns>
+    private bool EmitPublicEvents(IEnumerable<ProcessEvent> events)
+    {
+        bool eventsChanged = false;
+        events?.Where(stepEvent => stepEvent.Visibility == KernelProcessEventVisibility.Public).ToList().ForEach(stepEvent =>
```

Comment:
Coding / Perf Nit - Re-allocating as a list is a little expensive as opposed to simply using `foreach` on `IEnumerable`.  True, if you already have a list, the Linq extension `ForEach` has performance equivalent to a basic `for`...but if you have to re-allocate just to use the extension then I question the value.

Reply:
removed this code too